### ISSUE TITLE
Application-scoped Argo CD doesn't work when managed through Cluster-scoped Argo CD

### ISSUE THREAD

**Issue by nccurry** [redhat-developer/gitops-operator#221]
**Describe the bug**
When managing a secondary application-scoped ArgoCD instance through a primary ArgoCD instance, the namespace role bindings that the secondary application-scoped ArgoCD instance creates will get deleted by the primary ArgoCD instance.

**To Reproduce**

- Create a cluster-scoped ArgoCD instance
- Create a application-scoped ArgoCD instance
- Label a namespace with argocd.argoproj.io/managed-by=application-argocd
- Role bindings in the new namespace will get deleted by the cluster-scoped ArgoCD

This is because the application scoped ArgoCD instance will get deployed with a label that matches the cluster-scoped ArgoCD Application name

```
  labels:
    app.kubernetes.io/instance: <Cluster scoped Application name for deploying application scoped ArgoCD instance>
```

When you then annotate a namespace with ```argocd.argoproj.io/managed-by=application-argocd```, the rolebindings that get created will inherit the ```app.kubernetes.io/instance``` label on the application-scoped ArgoCD instance. 

This causes the cluster-scoped ArgoCD to pick them  up and delete then because ArgoCD picks up instance labels.
https://argoproj.github.io/argo-cd/faq/#why-is-my-app-out-of-sync-even-after-syncing
https://github.com/argoproj/argo-cd/issues/1482

**Expected behavior**
The ability to manage a secondary argocd instance from a primary argocd instance without conflict.

**Comment by iam-veeramalla** [comment]
Thanks @nccurry for reaching out. @shubhamagarwal19 I think we are fixing this issue in v1.3 right ?

**Comment by shubhamagarwal19** [comment]
@iam-veeramalla, yes! This is planned to go in v1.3 release!

**Comment by nccurry** [comment]
@shubhamagarwal19 @iam-veeramalla How does the fix work?

**Comment by jannfis** [comment]
@nccurry The fix is that we don't set the labels from the owning `ArgoCD` CR anymore on the resources we create.

**Comment by jannfis** [comment]
This is the fix in the upstream operator: https://github.com/argoproj-labs/argocd-operator/pull/414

**Comment by iam-veeramalla** [comment]
Fixed in latest - v1.3.0 and above
